src/mscorlib/src/System/Runtime/Intrinsics/X86/Sse2.cs

```diff
@@ -971,47 +972,136 @@ public static class Sse2
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<sbyte> SetVector128(sbyte e15, sbyte e14, sbyte e13, sbyte e12, sbyte e11, sbyte e10, sbyte e9, sbyte e8, sbyte e7, sbyte e6, sbyte e5, sbyte e4, sbyte e3, sbyte e2, sbyte e1, sbyte e0)
+        {
+            unsafe
+            {
+                sbyte* buffer = stackalloc sbyte[16];
+                buffer[0] = e0; buffer[1] = e1; buffer[2] = e2; buffer[3] = e3;
+                buffer[4] = e4; buffer[5] = e5; buffer[6] = e6; buffer[7] = e7;
+                buffer[8] = e8; buffer[9] = e9; buffer[10] = e10; buffer[11] = e11;
+                buffer[12] = e12; buffer[13] = e13; buffer[14] = e14; buffer[15] = e15;
+                return LoadVector128(buffer);
+            }
+        }
         /// <summary>
         /// ___m128i _mm_set_epi8 (char e15, char e14, char e13, char e12, char e11, char e10, char e9, char e8, char e7, char e6, char e5, char e4, char e3, char e2, char e1, char e0)
         ///   HELPER
         /// </summary>
-        public static Vector128<byte> SetVector128(byte e15, byte e14, byte e13, byte e12, byte e11, byte e10, byte e9, byte e8, byte e7, byte e6, byte e5, byte e4, byte e3, byte e2, byte e1, byte e0)  => SetVector128(e15, e14, e13, e12, e11, e10, e9, e8, e7, e6, e5, e4, e3, e2, e1, e0);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<byte> SetVector128(byte e15, byte e14, byte e13, byte e12, byte e11, byte e10, byte e9, byte e8, byte e7, byte e6, byte e5, byte e4, byte e3, byte e2, byte e1, byte e0)
+        {
+            unsafe
+            {
+                byte* buffer = stackalloc byte[16];
+                buffer[0] = e0; buffer[1] = e1; buffer[2] = e2; buffer[3] = e3;
+                buffer[4] = e4; buffer[5] = e5; buffer[6] = e6; buffer[7] = e7;
+                buffer[8] = e8; buffer[9] = e9; buffer[10] = e10; buffer[11] = e11;
+                buffer[12] = e12; buffer[13] = e13; buffer[14] = e14; buffer[15] = e15;
+                return LoadVector128(buffer);
+            }
+        }
         /// <summary>
         /// __m128i _mm_set_epi16 (short e7, short e6, short e5, short e4, short e3, short e2, short e1, short e0)
         ///   HELPER
         /// </summary>
-        public static Vector128<short> SetVector128(short e7, short e6, short e5, short e4, short e3, short e2, short e1, short e0) => SetVector128(e7, e6, e5, e4, e3, e2, e1, e0);
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static Vector128<short> SetVector128(short e7, short e6, short e5, short e4, short e3, short e2, short e1, short e0)
+        {
+            Vector128<int> vector1 = ConvertScalarToVector128Int32(e0);
```

Comment:
Same, we should profile this and determine what the actual "best" sequence is.

Reply:
Thnx for review - I have created #16838 to track both performance validation and creation of performance tests for HW intrinsics.